modules/icgc-argo-workflows/prep_sample/resources/usr/bin/main.py

```diff
@@ -0,0 +1,355 @@
+                                tumour_normal_designation
+                            ])
+
+    read_group_info.update({'DS': description})
+
+    return read_group_info
+
+def main():
+    """
+    Python implementation of tool: seq-data-to-lane-fastq
+    """
+
+    parser = ArgumentParser(description='Tool: seq-data-to-lane-fastq')
+    parser.add_argument("-s", "--input-files", dest="input_files", required=True,
+                        help="Input files to process", type=str, nargs='+')
+    parser.add_argument("-p", "--metadata-json", dest="metadata_json", required=True,
+                        help="JSON file containing song analysis")
+    parser.add_argument('-t', '--tempdir', dest='tempdir', type=str, default=".",
+                        help='Specify directory for temporary files')
+    parser.add_argument('-o', '--outdir', dest='outdir', type=str, default="out",
+                        help='Specify directory for output files')
+    parser.add_argument("-n", "--cpus", type=int, default=cpu_count())
+
+    args = parser.parse_args()
+
+    if os.path.exists(args.outdir) and os.path.isdir(args.outdir):
+      shutil.rmtree(args.outdir)
+    os.mkdir(args.outdir)
+
+    with open(args.metadata_json, 'r') as f:
+      song_analysis = json.load(f)
+
+    study_id = song_analysis['studyId']
+    donor_id = song_analysis['samples'][0]['donor']['donorId']
+    sample_id = song_analysis['samples'][0]['sampleId']
+    specimen_id = song_analysis['samples'][0]['specimen']['specimenId']
+    sex = 'XX' if song_analysis['samples'][0]['donor']['gender'] == 'Female' else 'XY' 
+    specimen_type = song_analysis['samples'][0]['specimen']['specimenType']
+    tumour_normal_designation = song_analysis['samples'][0]['specimen']['tumourNormalDesignation']
+    status = '0' if tumour_normal_designation == 'Normal' else '1'
+    analysis_type = song_analysis['analysisType']['name']
+    output_sample_sheet = f'{args.outdir}/{sample_id}_{analysis_type}_sample_sheet.csv'
+    
+
+    sample_sheet = dict()
+    if analysis_type == 'sequencing_experiment':
+      read_group_count = song_analysis['read_group_count']
+      filepair_map_to_readgroup = group_readgroup_by_filepair(song_analysis)
+
+      for fp in filepair_map_to_readgroup:
+        if filepair_map_to_readgroup[fp]['format'] == 'BAM':
+          # for bam just need fp[0] since fp[1] is either the same as fp[0] or None
+          sample_sheet = generate_fastqs_from_bam(filename_to_file(fp, args.input_files)[0],
+                                      filepair_map_to_readgroup[fp]['read_groups'],
+                                      args.cpus, sample_sheet, study_id, donor_id, sample_id, 
+                                      args.tempdir, args.outdir, specimen_id, specimen_type, 
+                                      tumour_normal_designation)
+        else: # FASTQ must be one read group
+          fq_pair = filename_to_file(fp, args.input_files)
+          fastq_pair = bunzip2(fq_pair)
```

Comment:
Yes. this is the step to handle `bz2` compressed file. We will still need this step, right? Because the downstream analysis only take `fastq` not `bz2` as input.

Reply:
Oh my assumption was that downstream tools like FASTQC or BWA-mem handled `bz2` natively , so no decompression was required. Okay this can be resolved then.